modules/src/ics23_commitment/specs.rs

```diff
@@ -0,0 +1,44 @@
+use ics23::ProofSpec;
+
+use ibc_proto::ics23::ProofSpec as ProtoProofSpec;
+
+/// An array of proof specifications.
+///
+/// This type encapsulates different types of proof specifications, mostly predefined, e.g., for
+/// Cosmos-SDK.
+/// Additionally, this type also aids in the conversion from `ProofSpec` types from crate `ics23`
+/// into proof specifications as represented in the `ibc_proto` type; see the
+/// `From` trait(s) below.
+pub struct ProofSpecs {
+    specs: Vec<ProofSpec>,
+}
+
+impl ProofSpecs {
+    /// Returns the specification for Cosmos-SDK proofs
+    pub fn cosmos() -> Self {
+        Self {
+            specs: vec![
+                ics23::iavl_spec(),       // Format of proofs-iavl (iavl merkle proofs)
+                ics23::tendermint_spec(), // Format of proofs-tendermint (crypto/ merkle SimpleProof)
+            ],
+        }
+    }
+}
+
+/// Converts from the domain type (which is represented as a vector of `ics23::ProofSpec`
+/// to the corresponding proto type (vector of `ibc_proto::ProofSpec`).
+/// TODO: fix with https://github.com/informalsystems/ibc-rs/issues/853
+impl From<ProofSpecs> for Vec<ProtoProofSpec> {
+    fn from(domain_specs: ProofSpecs) -> Self {
+        let mut raw_specs = vec![];
+        for ds in domain_specs.specs.iter() {
+            // Both `ProofSpec` types implement trait `prost::Message`. Convert by encoding, then
+            // decoding into the destination type.
```

Comment:
It would be nice to have a `SAFETY` not here describing why the conversion between the two proto types is infallible.

Reply:
That's a good idea. Let me know if you had something else in mind! f33bd51c

Hopefully we can get rid of this conversion altogether once we fix the associated issue (#853).